### ISSUE TITLE
[BUG/ISSUE] Unable to download from geoschemdata.computecanada.ca

### ISSUE THREAD

**Issue by bsuther** [geoschem/geos-chem#758]
Hello, 

I have been trying to download input files from geoschemdata.computecanada.ca unsuccessfully for several days now. It seems like the connection is just so slow that it gets reset before the files can be downloaded. I am wondering if there is an issue with the connection on the side of the server? I have tried downloading with a couple different internet connections on my end and none seemed to work. If you could help me figure out what I am doing wrong, or in the case that it is not an issue on my end if there is anything I can do, I would greatly appreciate it. Thank you!

 Here is the command I am using to try to download the data

>  wget -r -np -nH -N -R "*html" "geoschemdata.computecanada.ca/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/"

and here is an example of the output I am getting with wget

> --2021-06-23 14:14:00--  http://geoschemdata.computecanada.ca/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc
> Reusing existing connection to geoschemdata.computecanada.ca:80.
> HTTP request sent, awaiting response... 200 OK
> Length: 733023288 (699M) [application/x-netcdf]
> Saving to: ‘ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc’
> 
> 10% [===>                                   ] 78,365,512  41.7KB/s   in 33m 21s
> 
> 2021-06-23 14:47:25 (38.2 KB/s) - Connection closed at byte 78365512. Retrying.
> 
> --2021-06-23 14:47:26--  (try: 2)  http://geoschemdata.computecanada.ca/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc
> Connecting to geoschemdata.computecanada.ca (geoschemdata.computecanada.ca)|206.12.88.212|:80... connected.
> HTTP request sent, awaiting response... 206 Partial Content
> Length: 733023288 (699M), 654657776 (624M) remaining [application/x-netcdf]
> Saving to: ‘ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc’
> 
> 43% [++++===========>                       ] 315,703,112  732B/s   in 78m 50s 
> 
> 2021-06-23 16:06:27 (49.0 KB/s) - Connection closed at byte 315703112. Retrying.
> 
> --2021-06-23 16:06:29--  (try: 3)  http://geoschemdata.computecanada.ca/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc
> Connecting to geoschemdata.computecanada.ca (geoschemdata.computecanada.ca)|206.12.88.212|:80... connected.
> HTTP request sent, awaiting response... 206 Partial Content
> Length: 733023288 (699M), 417320176 (398M) remaining [application/x-netcdf]
> Saving to: ‘ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc’
> 
> 54% [++++++++++++++++====>                  ] 396,322,472  670KB/s   in 22m 40s
> 
> 2021-06-23 16:29:37 (57.9 KB/s) - Connection closed at byte 396322472. Retrying.
> 
> --2021-06-23 16:29:40--  (try: 4)  http://geoschemdata.computecanada.ca/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc
> Connecting to geoschemdata.computecanada.ca (geoschemdata.computecanada.ca)|206.12.88.212|:80... connected.
> HTTP request sent, awaiting response... 206 Partial Content
> Length: 733023288 (699M), 336700816 (321M) remaining [application/x-netcdf]
> Saving to: ‘ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2014/05/GEOSFP.20130717.A3cld.025x03125.NA.nc’
> 
> 65% [+++++++++++++++++++++++++++++++++++++=======>                        ] 481,200,352 29.2KB/s   in 45m 39s
> 
> 2021-06-23 17:15:28 (30.3 KB/s) - Connection closed at byte 481200352. Retrying.
>

**Comment by yantosca** [comment]
HI @bsuther, thanks for writing.  @LiamBindle was saying that there has been a lot of traffic on the Compute Canada server so it might just be swamped.  Also, the data is being migrated over to a new server (AWS-based) at Wash U in St. Louis soon, so hopefully that will improve reliability.

**Comment by eamarais** [comment]
@yantosca and @LiamBindle I've been experiencing the same issue for a few weeks now. I worked around this by managing to find the met fields I needed on the Harvard FTP. But, there is limited data on the Harvard ftp and now we're unable to download GEOS-FP native resolution files for Jan-Jul 2020.  Is there an interim solution?

**Comment by LiamBindle** [comment]
Hi @eamarais, sorry about that. The geoschemdata.computecanada.ca VM has been struggling a lot lately. I will ping the sysadmin and see if there's anything that can be done immediately. I'll follow up with what I hear.

The solution in the works is a new server with a higher capacity. The new server is being provisioned, and we're hopeful it will be online in the next week or two. I have a meeting with the sysadmins tomorrow, so I'll follow up with any more details regarding the timeline tomorrow afternoon.

**Comment by eamarais** [comment]
@LiamBindle Thanks for the update and for following up on this. A 1-2 week wait time doesn't sound too bad.

**Comment by LiamBindle** [comment]
Hi @eamarais. I'm told the new server should be live for testing next Friday (that was the estimate I heard this afternoon). The new address is going to be [https://geoschemdata.ris.wustl.edu](https://geoschemdata.ris.wustl.edu). 

The ComputeCanada admins say they identified two issues: one is fixed (something to do with their network), and the other fix is waiting for a system reboot. That reboot should happen today or Monday. Hopefully after the reboot the [https://geoschemdata.comptuecanada.ca](https://geoschemdata.comptuecanada.ca) issues will be fixed.

**Comment by bsuther** [comment]
That is all encouraging. Thank you all for your help!

**Comment by LiamBindle** [comment]
@bsuther, no problem! 

Thanks for everyone's patience.

**Comment by LiamBindle** [comment]
@bsuther @eamarais The reboot happened Friday. I just tried the ComputeCanada server and it's responding OK, but it is still pretty slow.

**Comment by tessac2** [comment]
@bsuther @LiamBindle I tried again yesterday on ComputeCanada and got similar errors--the connection kept timing out and when it did download data it was quite slow. 

Will there be instructions on how to connect to the new server address through WUSTL when it is ready?

**Comment by LiamBindle** [comment]
@tessac2 sorry the errors have persisted. I've asked the sysadmins to take another look at it. I'll follow up with any updates I hear.

To use the new server all you will need to do is change your URL to [https://geoschemdata.ris.wustl.edu](https://geoschemdata.ris.wustl.edu). It should look and feel identical to the ComputeCanada mirror, just faster and more reliable.

**Comment by eamarais** [comment]
Hi @LiamBindle, thanks for sharing the updated URL. Is it live now? I get a "This site can't be reached" error when I try access it from Chrome and a "Name or service not known" error on our servers.

**Comment by kilicomu** [comment]
+1 on trouble with Compute Canada's data server! It has been intermittently okay this week, but new system sounds promising.

**Comment by YanshunLi-washu** [comment]
@eamarais Hi, the engineer in washu is still working on that new site. We'll inform you once it's ready. Hope you understand.

**Comment by LiamBindle** [comment]
Hi all. Sorry, I was on vacation last week. I'm disappoint to hear the new server isn't online. I just pinged the sysadmins and they said they are hoping to have it online by Wednesday. I'm not sure what the delay was, but it sounds like they're doing their internal code review today. I'll follow up Wednesday if it still isn't online.

**Comment by LiamBindle** [comment]
It looks like it still isn't online. I'm going to ping the sysadmins and see where it's at.

**Comment by LiamBindle** [comment]
Hi all. A couple threads came together today.

**tldr:**
* I created a new globus endpoint called "GEOS-Chem Data (WashU)". You can download data from there. 
* The timescale for a https://geoschemdata.computecanada.ca fix is 1-2 weeks, but it might not be fixable. 
* The timescale for https://geoschemdata.ris.wustl.edu to go live is ~2 weeks

### Status of https://geoschemdata.computecanada.ca
There was using so much bandwidth that it was saturating ComputeCanada's network and negatively affecting other projects. Therefore, the admins implemented a 80 Mbps throttler which they say we're continuously hitting. The admins suspect that some of this traffic is illegitimate. 

Possible interim solution: We are talking with ComputeCanada about implementing an allow-list of legitimate IPs. This is a pain and far from ideal, but at least that way we can make sure only legitimate users are using that 80 Mbps. If the admins say this is doable I'll start collecting a list of IPs that want to use this server. This server will hopefully stay live for another 2-3 months.

### Status of https://geoschemdata.ris.wust.edu
The admins ran into some issues in the backend that has delayed their timeline. Their new estimate is 2 weeks, but it sounded like a guess.

### Interim solution: A new Globus endpoint
I was able to set up a Globus endpoint for the GEOS-Chem data hosted at WashU. This is the same directory that the new FTP site will be sharing. The endpoint is named "GEOS-Chem Data (WashU)".
![image](https://user-images.githubusercontent.com/12772306/126013490-307aae51-8533-44fc-ab0f-c31dc0d746d1.png)

To access this data you will need a globus endpoint (that's Globus's terminology for the root directory where data can be transfered to/from). If your cluster doesn't have a globus endpoint you can create your own by following these instructions: https://docs.globus.org/how-to/globus-connect-personal-linux/.

**Comment by LiamBindle** [comment]
The ComputeCanada admins found two IPs that appeared to be DDoSing the server. They blocked those IPs yesterday. To me it seems that made https://geoschemdata.computecanada.ca quite a bit more responsive.

**Comment by LiamBindle** [comment]
The admins found a third suspicious IP. Since the three IPs were blocked I've heard from several users that the speeds are better (a few Mbps). A few Mbps isn't great but hopefully it's enough to hold us over until the WashU server is live. If anyone is still getting really bad speeds, please let me know.

On Friday the WashU admins said they solved the issue that delayed it's release a few weeks back and that now they are in the final development stages (DNS). They estimate Friday or early next week as the date it will go live.

**Comment by eamarais** [comment]
Thanks @LiamBindle for the update. The download speeds are much better now. It seems to be behaving as it was a few months ago.

**Comment by LiamBindle** [comment]
@eamarais Thanks for letting me know! Glad to hear things are better.

**Comment by LiamBindle** [comment]
Hi all. Quick update on this issue.

https://geoschemdata.computecanada.ca should be back to normal. It's limited to 80 Mbps egress, so slowness is still possible depending on how many people are trying to access it. Let us know if it's bad for a couple days in a row---it's possible it's getting spammed by bots and we need to ban some IPs. 

https://geoschemdata.wustl.edu is live but we're still working on updating metfields. Emissions, chemistry inputs, restart files, etc. should be good though. This site should be considered a "beta" version for the time being. 

If you use Globus, the "GEOS-Chem Data (WashU)" endpoint should be preferred. It's much faster than the ComputeCanada endpoint which is also subjec to share that 80 Mbps limit. Similar to the WUSTL site we're still working on updating metfields but emissions, chemistry inputs, restart files, etc should be okay.

Thanks @YanshunLi-washu for all your work on updating the metfields!

I'm going to close this issue since things are back to normal. We'll probably announce the https://geoschemdata.wustl.edu site in the near future once we're sure that all the data is present and valid.

Thanks for everyones help and patience.

**Comment by bsuther** [comment]
Hello, I was wondering if anyone else is now having similar issues with the wustl.edu site now? I have been trying unsuccessfully to download some files for a few days now. Also, when I try to check for the files by going to https://geoschemdata.wustl.edu/ in a web browser I get a "This site can’t be reached" message. Thanks.

**Comment by yantosca** [comment]
There have been recently some issues on the wustl FTP site, and I believe it is being looked into.  If you have access to Globus Endpoint, then you can download from the endpoint "GEOS Chem Data (WashU)".

**Comment by LiamBindle** [comment]
Hi all, I'm reopening this. I'm getting the same as @bsuther. I opened a ticket with the administrators. Sorry for the troubles. I'll follow up when I know something.

**Comment by LiamBindle** [comment]
Hi all, http://geoschemdata.wustl.edu/ should be back online. Can you try it now? Sorry for the delay in getting it back online.

**Comment by kilicomu** [comment]
Looks better to me!

**Comment by bsuther** [comment]
Thank you @LiamBindle for getting the server back up and running, is looks like it is working from my end too. Thank you for the suggestion @yantosca, I was able to get Globus set up and was able to get what I needed that way in the meantime.

**Comment by yantosca** [comment]
I think we can close out this issue.  The geoschemdata.wustl.edu data server is online, if you are having difficulties, you can use Globus, or dry downloading from geoschemdata.computecanada.ca (which is currently online but will eventually be deactivated).
